[hipTensor] [Cherry-Pick] Bump version to 2.2.0 and update CHANGELOG.md for ROCm 7.2 (#2845)
ROCm/rocm-libraries: projects/hiptensor/CHANGELOG.md

Full documentation for hipTensor is available at [rocm.docs.amd.com/projects/hiptensor](https://rocm.docs.amd.com/projects/hipTensor/en/latest/index.html).

> ## (Unreleased) hipTensor 2.2.0 for ROCm 7.2.0

Reviewer: The `(Unreleased)` should be removed at this point.

Author: removed

* Added `hiptensorHandleReadPlanCacheFromFile` to read a plan cache from a file into a hipTensor handle.
* Added `simple_contraction_plan_cache` to demonstrate plan cache usages.
* Added `plan_cache_test` to test the plan cache across various tensor ranks.
* Added C API headers to enable compatibility with C programs.

> * Added cmake function to allow projects querying about architecture support.

Reviewer 2: ```suggestion
* Added a CMake function to allow projects to query architecture support.
```

Author: change it accordingly

> * Added option to configure memory layout for tests and benchmarks.

Reviewer: The memory layout option didn't make into the release branch. Can you remove it from here?

Author: The memory layout is already in the release branch, please check https://github.com/ROCm/rocm-libraries/commits/release/rocm-rel-7.2/projects/hiptensor

Reviewer: From the internal information I had received, the 7.2 branch would have been created from before the complete integration of hipTensor in rocm-libraries, without including this change. And in that case we'd have a 7.2 branch in the hipTensor repo and would have to apply these changelog modifications to that repo instead. But you're right that memory layout changes are already in the release branch as well as hipTensor is in rocm-libraries.

> * Added option to configure memory layout for tests and benchmarks.

Reviewer 2: ```suggestion
* Added an option to configure the memory layout for tests and benchmarks.
```

Author: change it accordingly

### Changed

* Updated C++ standard from C++17 to C++20.
* Include files `hiptensor/hiptensor.hpp` and `hiptensor/hiptensor_types.hpp` are now deprecated. Use `hiptensor/hiptensor.h` and `hiptensor/hiptensor_types.h` instead.
* Converted include guards from #ifndef/#define/#endif to #pragma once.

### Resolved issues

* Removed large tensor sizes causing problem in benchmarks.

## hipTensor 2.1.0 for ROCm 7.1.0

### Added

> * Added large tensor lengths in benchmark yaml files.

Reviewer 2: ```suggestion
* Added large tensor lengths in benchmark YAML files.
```

Author: changed it accordingly
